Geometry configuration check (issue #24328) [2/10]

PetrilloAtWork opened the pull request:
This is part of the pull request set including also LArSoft/larcore#6.
The main discussion will happen in that pull request.

PetrilloAtWork committed:
Added data product with geometry information for configuration checks.

This is part of the resolution of issue #24328.

FNALbuild commented:
A new Pull Request was created by  @PetrilloAtWork (Gianluca Petrillo) for develop.

It involves the following packages:

larcoreobj


@LArSoft/level-1-managers, @LArSoft/level-2-managers can you please review it and eventually sign? Thanks.

cms-bot commands are listed <a href="https://cdcvs.fnal.gov/redmine/projects/larsoft/wiki/Pull_request_comments_that_trigger_CI_actions">here</a>

FNALbuild commented:
The code-checks are being triggered in jenkins.

knoepfel commented on larcoreobj/SummaryData/GeometryConfigurationInfo.h:
```diff
@@ -0,0 +1,83 @@
+  
+  
+  /// Protocol: whether the data content is valid.
+  bool isDataValid() const { return dataVersion != InvalidDataVersion; }
```
Can be marked `noexcept`.

PetrilloAtWork replied:
I keep forgetting to.

knoepfel commented on larcoreobj/SummaryData/GeometryConfigurationInfo.cxx:
```diff
@@ -0,0 +1,55 @@
+{
+  
+  if (info.isDataValid()) {
+    out <<   "Geometry information version: " << info.dataVersion;
```
To avoid extended nesting,  you might consider:

```.cpp
if (not info.isDataValid()) {
  return os << "Invalid geometry configuration information" << std::endl;
}

if (info.dataVersion  >= sumdata::GeometryConfigurationInfo::DataVersion_t{1}) ...
```

PetrilloAtWork replied:
Hmm... ok. Whatever reason I may have had not to do it, it was weak enough that I don't remember it any more.

knoepfel commented on larcoreobj/SummaryData/GeometryConfigurationInfo.cxx:
```diff
@@ -0,0 +1,55 @@
+    
+    out
+      << "\nFull configuration:"
+      << "\n" << std::string(80, '-')
+      << "\n" << info.geometryServiceConfiguration
+      << "\n" << std::string(80, '-')
```
There is a _cetlib_ facility for making horizontal rules.
```.cpp
#include "cetlib/HorizontalRule.h"

void print() 
{
  cet::HorizontalRule const rule{8}; // Number of repeated characters
  std::cout << rule('+') << '\n'  // prints "++++++++"
            << rule('-') << '\n'; // prints "--------"
}
```
Maybe overkill for what you need, though.

PetrilloAtWork replied:
I was not aware of it.
I will not adopt it in this case because I feel it's not worth the additional dependency — right now, the library is pure C++ and Standard Library.
(ok, the dictionary library pulls in a load of dependencies — MessageFacility?!? — so I guess the argument is not as strong as I thought)

knoepfel replied:
I agree there is no need to introduce the explicit dependency here.

knoepfel reviewed (commented):
Looks good, Gianluca.  Just some very minor comments.

PetrilloAtWork committed:
Added `noexcept` specifier.

FNALbuild commented:
Pull request #9 was updated. @LArSoft/level-1-managers, @LArSoft/level-2-managers can you please check and sign again.

FNALbuild commented:
The code-checks are being triggered in jenkins.

FNALbuild commented:
+code-checks  
 for details see 
 https://buildmaster.fnal.gov/buildmaster/job/GithubIntegration/job/run-pr-code-checks//779/console

PetrilloAtWork committed:
Added support for version 1 of geo::GeometryConfigurationInfo.

What was version 1 is now version 2.
Also rearranged a bit the structure of the object content printout.

FNALbuild commented:
Pull request #9 was updated. @LArSoft/level-1-managers, @LArSoft/level-2-managers can you please check and sign again.

FNALbuild commented:
Pull request #9 was updated. @LArSoft/level-1-managers, @LArSoft/level-2-managers can you please check and sign again.

FNALbuild commented:
The code-checks are being triggered in jenkins.

FNALbuild commented:
The code-checks are being triggered in jenkins.

FNALbuild commented:
+code-checks  
 for details see 
 https://buildmaster.fnal.gov/buildmaster/job/GithubIntegration/job/run-pr-code-checks//780/console

knoepfel commented:
Will approve as soon as there is L2 approval.

absolution1 commented:
approve

FNALbuild commented:
The tests are being triggered in jenkins.

FNALbuild commented:
This pull request is fully signed and it will be merged to develop and built in the next LArSoft release after it passes the integration tests. This pull request will now be reviewed by the release team before it's merged.)

FNALbuild commented:
-LArSoft tests failed on slf7 for c7:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/10187&builds= 
 -build

FNALbuild commented:
+LArSoft tests OK on slf7 for e19:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/10186&builds= 
 +build

FNALbuild commented:
-lariat failed on slf7 for e19:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:lariat/view_builds/index?offset=0&builds=lariat_ci/5121&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/10186&builds=

FNALbuild commented:
+argoneut tests OK on slf7 for e19:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:argoneut/view_builds/index?offset=0&builds=argoneut_ci/5323&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/10186&builds=

FNALbuild commented:
+sbnd tests OK on slf7 for e19:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:sbnd/view_builds/index?offset=0&builds=sbnd_ci/5790&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/10186&builds=

FNALbuild commented:
-icarus tests warning on slf7 for e19:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:icarus/view_builds/index?offset=0&builds=icarus_ci/1388&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/10186&builds=

FNALbuild commented:
-dune tests warning on slf7 for e19:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:dune/view_builds/index?offset=0&builds=dune_ci/10570&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/10186&builds=

FNALbuild commented:
-uboone tests warning on slf7 for e19:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:uboone/view_builds/index?offset=0&builds=uboone_ci/5746&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/10186&builds=

lgarren commented:
GeometryConfigurationInfo is forward declared as a class, but is officially declared as a struct.  I'm not sure why no one caught that.  My local clang build just complained.  I have changed the forward declaration to struct and it will go into the release that way.

PetrilloAtWork commented:
Thank you, Lynn. This inconsistency is detected by Clang but not by GCC... I don't remember if I ever tried compilation with Clang.
